# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderScrollData.cpp
# Commit: 96da9d241051
# Full Hash: 96da9d241051d222bdd693bdb519d058a013b59c
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-23 09:46:24
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Core renderroot splitting changes r=kats,sotaro
#   
#   This is a large patch that contains all of the core changes for
#   renderroot splitting.
#   
# ==============================================================================

diff -r 8cefa694f811 -r 96da9d241051 gfx/layers/wr/WebRenderScrollData.cpp
--- a/gfx/layers/wr/WebRenderScrollData.cpp	Fri Mar 22 18:28:31 2019 +0000
+++ b/gfx/layers/wr/WebRenderScrollData.cpp	Fri Mar 22 18:28:42 2019 +0000
@@ -21,7 +21,8 @@
     : mDescendantCount(-1),
       mTransformIsPerspective(false),
       mEventRegionsOverride(EventRegionsOverride::NoOverride),
-      mFixedPosScrollContainerId(ScrollableLayerGuid::NULL_SCROLL_ID) {}
+      mFixedPosScrollContainerId(ScrollableLayerGuid::NULL_SCROLL_ID),
+      mRenderRoot(wr::RenderRoot::Default) {}
 
 WebRenderLayerScrollData::~WebRenderLayerScrollData() {}
 
@@ -32,11 +33,13 @@
 void WebRenderLayerScrollData::Initialize(
     WebRenderScrollData& aOwner, nsDisplayItem* aItem, int32_t aDescendantCount,
     const ActiveScrolledRoot* aStopAtAsr,
-    const Maybe<gfx::Matrix4x4>& aAncestorTransform) {
+    const Maybe<gfx::Matrix4x4>& aAncestorTransform,
+    wr::RenderRoot aRenderRoot) {
   MOZ_ASSERT(aDescendantCount >= 0);  // Ensure value is valid
   MOZ_ASSERT(mDescendantCount ==
              -1);  // Don't allow re-setting an already set value
   mDescendantCount = aDescendantCount;
+  mRenderRoot = aRenderRoot;
 
   MOZ_ASSERT(aItem);
   aItem->UpdateScrollData(&aOwner, this);
@@ -188,10 +191,6 @@
   return (it == mScrollIdMap.end() ? Nothing() : Some(it->second));
 }
 
-void WebRenderScrollData::SetFocusTarget(const FocusTarget& aFocusTarget) {
-  mFocusTarget = aFocusTarget;
-}
-
 void WebRenderScrollData::SetIsFirstPaint() { mIsFirstPaint = true; }
 
 bool WebRenderScrollData::IsFirstPaint() const { return mIsFirstPaint; }